packages/mdc-ripple/foundation.js

```diff
@@ -488,20 +488,31 @@ class MDCRippleFoundation extends MDCFoundation {
     }
   }
 
-  layout() {
+  /**
+   * Recomputes all dimensions and positions for the ripple element.
+   * Optionally include width and height for ripple surfaces that change size during runtime.
+   * @param {number=} width
+   * @param {number=} height
+   */
+  layout(width, height) {
     if (this.layoutFrame_) {
       cancelAnimationFrame(this.layoutFrame_);
     }
     this.layoutFrame_ = requestAnimationFrame(() => {
-      this.layoutInternal_();
+      this.layoutInternal_(width, height);
       this.layoutFrame_ = 0;
     });
   }
 
-  /** @private */
-  layoutInternal_() {
+  /**
+   * @param {number=} width
+   * @param {number=} height
+   * @private
+   */
+  layoutInternal_(width, height) {
```

Comment:
I suspect that we should accept a single object that includes a width and height, and assign that to `this.frame_` when present instead of calling `computeBoundingRect`.

* `this.frame_` is also used in `updateLayoutCssVars_` for unbounded ripples, and in `getFgTranslationCoordinates_` for ripples activated via keyboard, and we would want the updated values to be used in those functions as well. (This also suggests to me that we need more tests to verify that the correct values are used when layout is called with an argument...)
* `layout` should really be called either with both width and height, or neither, and in the case where the argument is passed, we can save the browser from needing to layout/reflow.